SofaKernel/modules/SofaHelper/src/sofa/helper/AdvancedTimer.cpp

```diff
@@ -1406,8 +1406,27 @@ std::map<AdvancedTimer::IdStep, StepData> AdvancedTimer::getStepData(IdTimer id,
 helper::vector<Record> AdvancedTimer::getRecords(IdTimer id)
 {
     TimerData& data = timers[id];
-    for (unsigned int i=0; i<data.records.size(); ++i)
-        data.records[i].label = std::string(AdvancedTimer::IdStep(data.records[i].id));
+    for (Record & r : data.records) {
+        switch (r.type) {
+            case Record::RBEGIN: // Timer begins
+            case Record::REND: // Timer ends
+                r.label = (std::string) AdvancedTimer::IdTimer(r.id);
```

Comment:
use use std::string(... ) instead of (std::string)... ?

Reply:
It is exactly the same, see [here](https://stackoverflow.com/questions/45505861/function-style-cast-vs-constructor) and [here](https://web.archive.org/web/20070929132736/http://www.acm.org/crossroads/xrds3-1/ovp3-1.html). Apparently, since we are in c++, the good way to do it would be static_cast (which again, produce the exact same thing). I'll change it with static_cast if that's ok with you ;)